packages/storage/src/implementation/list.ts

```diff
@@ -24,26 +24,41 @@ import * as json from './json';
 import * as type from './type';
 import { ListResult } from '../list';
 
-export function fromResource(
+interface ListMetadataReource {
+  name: string;
+  bucket: string;
+}
+
+interface ListResultResource {
+  prefixes: string[];
+  items: ListMetadataReource[];
+  nextPageToken?: string;
+}
+
+const maxResultsKey = 'maxResults';
+const pageTokenKey = 'pageToken';
+const prefixesKey = 'prefixes';
+const itemsKey = 'items';
+
+function fromResource(
   authWrapper: AuthWrapper,
-  resource: { [name: string]: any }
+  resource: ListResultResource
```

Comment:
Is "resource" really a response here? Please push back if "resource" is the expected terminology here, but I think we could still improve our terminology here a bit. 

```
function convertBackendResponse(authWrapper: AuthWrapper, response: ListResponse) : ListResult {...}
```